src/more_ops.rs

```diff
@@ -180,9 +180,12 @@ pub fn op_divmod<T: Allocator>(args: &Node<T>) -> Response<T::Ptr> {
         let q = &a0 / &a1;
         let r = &a0 - &a1 * &q;
 
+        let signed_quotient = (a0.sign() == Sign::Minus || a1.sign() == Sign::Minus)
+            && a0.sign() != a1.sign();
```

Comment:
the issue was that "zero` does not have a sign (according to `sign()`) so even though a / b was negative, `q` would not be. So this logic computes the sign explicitly, without relying on it propagating to `q`